spec/models/post_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject { Post.new(title: 'My First Post', comments_counter: 4, likes_counters: 20) }
+
+  before { subject.save }
+
+  it 'title should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'comments_counter should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes counter should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'title should not be to long' do
+    subject.title = 'a' * 256
+    expect(subject).to_not be_valid
+  end
+
+  it 'comments counter should be integer' do
+    subject.comments_counter = 'b'
+    expect(subject).to_not be_valid
+  end
+
+  it 'comments counter should be greater than or equal to zero' do
+    subject.comments_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes counter should be integer' do
+    subject.likes_counters = 'c'
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes counter should be greater than or equal to zero' do
+    subject.likes_counters = -2
```

Comment:
- [ ] According to the project requirement, 👇🏻 
1. You should write unit tests for all your models: `User`, `Post`, `Comment`, and `Like` models. Kindly add unit tests for the  `Comment`, and `Like` models.
2. Any custom methods in the models should have a corresponding unit test. You're missing adding unit tests of the  `post_counter and recent_comments` methods of your Post model, the ` comment_counter` method of your Comment model, and the `like_counter` method. Please address this ✍🏻 

![im2](https://user-images.githubusercontent.com/79058364/215210008-f647c979-9f49-40cc-a73e-e5d3c29cfbb1.png)